src/spikeinterface/generation/splitting_tools.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+from spikeinterface.core.numpyextractors import NumpySorting
+
+
+def split_sorting_by_times(
+    sorting_analyzer, splitting_probability=0.5, partial_split_prob=0.95, unit_ids=None, min_snr=None, seed=None
+):
+    """
+    Fonction used to split a sorting based on the times of the units. This
+    might be used for benchmarking meta merging step (see components)
+
+    Parameters
+    ----------
+    sorting_analyzer : A sortingAnalyzer object
+        The sortingAnalyzer object whose sorting should be splitted
+    splitting_probability : float, default 0.5
+        probability of being splitted, for any cell in the provided sorting
+    partial_split_prob : float, default 0.95
+        The percentage of spikes that will belong to pre/post splits
+    unit_ids : list of unit_ids, default None
+        The list of unit_ids to be splitted, if prespecified
+    min_snr : float, default=None
+        If specified, only cells with a snr higher than min_snr might be splitted
+    seed : int | None, default: None
+        The seed for random generator.
+
+    Returns
+    -------
+    new_sorting, splitted_pairs : The new splitted sorting, and the pairs that have been splitted
+    """
+
+    sa = sorting_analyzer
+    sorting = sa.sorting
+    rng = np.random.RandomState(seed)
+
+    nb_splits = int(splitting_probability * len(sorting.unit_ids))
+    if unit_ids is None:
+        select_from = sorting.unit_ids
+        if min_snr is not None:
+            if sa.get_extension("noise_levels") is None:
+                sa.compute("noise_levels")
+            if sa.get_extension("quality_metrics") is None:
+                sa.compute("quality_metrics", metric_names=["snr"])
+
+            snr = sa.get_extension("quality_metrics").get_data()["snr"].values
+            select_from = select_from[snr > min_snr]
+
+        to_split_ids = rng.choice(select_from, nb_splits, replace=False)
+    else:
+        to_split_ids = unit_ids
+
+    spikes = sa.sorting.to_spike_vector()
+    new_spikes = spikes.copy()
+    max_index = np.max(spikes["unit_index"])
+    new_unit_ids = list(sa.sorting.unit_ids.copy())
+    splitted_pairs = []
+    for unit_id in to_split_ids:
+        ind_mask = spikes["unit_index"] == sa.sorting.id_to_index(unit_id)
```

Comment:
This is very slow you should use spike_vector_to_indices() from sorting tools.py

Reply:
This has beeen fixes, and tests have been added